### PR TITLE
Add provision to specify the crypt hash

### PR DESCRIPTION
**Pull request by nmahabaleshwar**
Add enum's to specify the crypt hash while rotating the password via gnsi.credentialz

### CODE REVIEW

File: credentialz/credentialz.proto [hunk 1]
```diff
@@ -510,11 +510,24 @@ message UserPolicy {
 message AuthorizedUsersResponse {
 }
 
+// Hash type's
+enum HashType {
+  HASH_TYPE_UNSPECIFIED = 0;
+  HASH_TYPE_CRYPT_MD5 = 1;
+  HASH_TYPE_CRYPT_SHA_2_512 = 2;
+}
```
Comment (tomek-US) [inline credentialz/credentialz.proto:518]: Is this enum going to be used in other messages than the CryptoHash message? If not then please move it into it.

File: credentialz/credentialz.proto [hunk 2]
```diff
@@ -510,11 +510,24 @@ message UserPolicy {
...
+
+message CryptoHash {
+  // Specifies the hash type used and value
+  HashType hashType = 1;
+  string hashValue = 2;
+}
```
Comment (tomek-US) [inline credentialz/credentialz.proto:524]: Is this enum to be used in other messages than the Password message? If not then please move it inside it.

File: credentialz/credentialz.proto (hunk 1 above)
Comment (nmahabaleshwar) [reply credentialz/credentialz.proto]: Addressed this

File: credentialz/credentialz.proto (hunk 2 above)
Comment (nmahabaleshwar) [reply credentialz/credentialz.proto]: Addressed this

**Pull request merged**